src/wizards.ts

```diff
@@ -0,0 +1,250 @@
+  newWizardEvent,
+  SimpleAction,
+  Wizard,
+  WizardAction,
+} from './foundation.js';
+
+interface MergeOptions {
+  title?: string;
+  selected?: (diff: Diff<Element | string>) => boolean;
+}
+
+export type Diff<T> =
+  | { ours: T; theirs: null }
+  | { ours: null; theirs: T }
+  | { ours: T; theirs: T };
+
+function describe(element: Element): string {
+  const id = identity(element);
+
+  return typeof id === 'string' ? id.replace(/^>/, '') : get('unidentifiable');
+}
+
+function mergeWizardAction(
+  attrDiffs: [string, Diff<string>][],
+  childDiffs: Diff<Element>[],
+  sink: Element,
+  source: Element,
+  options?: MergeOptions
+): WizardAction {
+  return (_, wizard: Element): EditorAction[] => {
+    const actions: SimpleAction[] = [];
+    const checkList = wizard.shadowRoot!.querySelector('mwc-list')!;
+
+    const selectedAttrDiffs = (<ListItem[]>checkList.selected)
+      .filter(item => item.classList.contains('attr'))
+      .map(item => attrDiffs[(item.value as unknown) as number]);
+
+    if (selectedAttrDiffs.length) {
+      const newSink = <Element>sink.cloneNode(false);
+      if (sink.childElementCount === 0) newSink.textContent = sink.textContent;
+      for (const [name, diff] of selectedAttrDiffs)
+        if (name === 'value') {
+          newSink.textContent = diff.theirs;
+        } else if (diff.theirs === null) newSink.removeAttribute(name);
+        else newSink.setAttribute(name, diff.theirs);
+      actions.push({ old: { element: sink }, new: { element: newSink } });
+    }
+
+    let acted = false;
+
+    const selectedChildDiffs = (<ListItem[]>checkList.selected)
+      .filter(item => item.classList.contains('child'))
+      .map(item => childDiffs[(item.value as unknown) as number]);
+    if (selectedChildDiffs.length) {
+      for (const diff of selectedChildDiffs)
+        if (!diff.ours)
+          actions.push({
+            new: { parent: sink, element: diff.theirs, reference: null },
+          });
+        else if (!diff.theirs)
+          actions.push({
+            old: {
+              parent: sink,
+              element: diff.ours,
+              reference: diff.ours.nextElementSibling,
+            },
+          });
+        else {
+          acted = true;
+          wizard.dispatchEvent(
+            newWizardEvent(
+              mergeWizard(diff.ours, diff.theirs, {
+                ...options,
+                title: undefined,
+              })
+            )
+          );
+        }
+    }
+
+    if (actions.length === 0 && !acted) wizard.dispatchEvent(newWizardEvent());
+
+    return [
+      {
+        actions,
+        title: get('merge.log', {
+          sink: describe(sink),
+          source: describe(source),
+          tag: sink.tagName,
+        }),
+      },
+    ];
+  };
+}
+
+export function mergeWizard(
+  sink: Element,
+  source: Element,
+  options?: MergeOptions
+): Wizard {
+  const attrDiffs: [string, Diff<string>][] = [];
+  const ourText = sink.textContent ?? '';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/openscd/open-scd/pullRequest?prid=7195001)
```suggestion
  const ourText = sink.textContent ?? "";
```